sisl/geometry.py

```diff
@@ -111,7 +111,7 @@ class Geometry(SuperCellChild):
     Atom : contained atoms are each an object of this
     """
 
-    def __init__(self, xyz, atom=None, sc=None):
+    def __init__(self, xyz, atom=None, sc=None, names=None):
```

Comment:
Hmm... probably this should be `namedindex` but I need to think if the NamedIndex functionality could be extended, and if so, perhaps having them in the constructor is bad... :( ?
Well...

Reply:
I named it like so because the named index is currently accesible on `geom.names`. I think it's good to have them be named similarly?

Reply:
Yes, I know, but I haven't decided whether `names` is the appropriate inner name (also it is currently "hidden" so hence I haven't cleaned it.